runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -0,0 +1,300 @@
+     *
+     * To re-constitute the commit order in the presence of transactions the CorfuRecords returned
+     * here have their UUID fields composed of two parts:
+     *     +----------------------------------------------------------------------------+
+     *     | Stream snapshot |Index in snapshot |     ID of the entry in the map        |
+     *     +----------------------------------------------------------------------------+
+     *     <----5 bytes------><----3 bytes-----><------- 8 bytes ----------------------->
+     *
+     * <p>This function currently does not return a view like the java.util implementation,
+     * and changes to the entryList will *not* be reflected in the map. </p>
+     *
+     * @param maxEntries - Limit the number of entries returned from start of the queue
+     *                   -1 implies all entries
+     * @return List of Entries sorted by their enqueue order
+     */
+    @Accessor
+    public List<CorfuQueueRecord<E>> entryList(int maxEntries) {
+        if (maxEntries == -1) {
+            maxEntries = Integer.MAX_VALUE;
+        }
+
+        long snapshotVersion = 0;
+        if (TransactionalContext.getCurrentContext() != null) {
+            snapshotVersion = TransactionalContext.getCurrentContext().getSnapshotTimestamp().getSequence();
+        }
+        List<CorfuQueueRecord<E>> copy = new ArrayList<>(Math.min(mainMap.size(), maxEntries));
+        int index = 0;
+        for (Map.Entry<Long, E> entry : mainMap.entrySet()) {
+            if (++index >= maxEntries) {
+                break;
+            }
+            UUID orderedId = new UUID((snapshotVersion<<MAX_BITS_FOR_INDEX) | index,
+            entry.getKey());
+            CorfuQueueRecord<E> record = new CorfuQueueRecord<>(orderedId, entry.getValue());
+            copy.add(record);
+        }
+        return copy;
+    }
+
+    @Accessor
+    public boolean isEmpty() {
+        return mainMap.isEmpty();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Accessor
+    public boolean containsKey(UUID key) {
+        return containsKeyInternal(key.getLeastSignificantBits());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Accessor
+    protected boolean containsKeyInternal(@ConflictParameter Long key) {
+        return mainMap.containsKey(key);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @TransactionalMethod(readOnly = true)
+    public E get(UUID key) {
+        return getInternal(key.getLeastSignificantBits());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Accessor
+    protected E getInternal(@ConflictParameter Long key) {
+        return mainMap.get(key);
+    }
+
+    /**
+     * Removes a specific element identified by the ID returned via enqueue() or entryList()'s CorfuRecord.
+     *
+     * {@inheritDoc}
+     *
+     * @return The entry that was successfully removed or null if there was no mapping.
+     */
+    @TransactionalMethod
+    public E removeEntry(UUID id) {
```

Comment:
Need to annotate this with a `MutatorAccessor` and mark the parameter as a `@ConflictParameter`

Reply:
Is this method still needed?